manager/notifier.go

```diff
@@ -184,6 +188,42 @@ func (n *notifier) sendEmailNotification(to string, a *Alert) error {
 	}
 	defer c.Quit()
 
+	// Authenticate if we and the server are both configured for it.
+	hasAuth, mechs := c.Extension("AUTH")
+	username := os.Getenv("SMTP_AUTH_USERNAME")
+	if hasAuth && username != "" {
+		for _, mech := range strings.Split(mechs, " ") {
+			switch mech {
+			case "CRAM-MD5":
+				secret := os.Getenv("SMTP_AUTH_SECRET")
+				if secret == "" {
+					continue
+				}
+				if err := c.Auth(smtp.CRAMMD5Auth(username, secret)); err != nil {
+					return fmt.Errorf("cram-md5 auth failed: %s", err)
+				}
+				break
```

Comment:
Wouldn't this `break` only leave the `switch`? From the Go spec: 'A "break" statement terminates execution of the innermost "for", "switch", or "select" statement within the same function.'

I guess your intention is to break out of the `for` loop (not that it would matter in most cases), which could be done via a label?